### PR TITLE
ENH: Add `SpecificAssets` filter.

### PR DESCRIPTION
**Pull request by ssanderson**
Adds a filter that matches a set of assets.  Mainly useful for testing and debugging.

### CODE REVIEW

File: zipline/pipeline/filters/filter.py [hunk 1]
```diff
@@ -494,3 +496,22 @@ def _compute(self, arrays, dates, assets, mask):
...
+
+class SpecificAssets(Filter):
+    """
+    A Filter that computes True for a specific set of predetermined assets.
+    """
+    inputs = ()
+    window_length = 0
+    params = ('sids',)
+
+    @expect_types(assets=(list, tuple, np.ndarray))
```
Comment (llllllllll) [inline zipline/pipeline/filters/filter.py]: how do we get an ndarray, it looks like we will coerce it to a `list` in the coerce below.

File: zipline/pipeline/filters/filter.py [hunk 2]
```diff
@@ -494,3 +496,22 @@ def _compute(self, arrays, dates, assets, mask):
...
+    @coerce_types(assets=((list, np.ndarray, pd.Series), list))
+    def __new__(cls, assets):
+        sids = frozenset(asset.sid for asset in assets)
```
Comment (llllllllll) [inline zipline/pipeline/filters/filter.py]: why do we need to coerce this to a list to iterate over it?

File: tests/pipeline/test_filter.py [hunk 3]
```diff
@@ -820,3 +822,40 @@ def test_top_and_bottom_with_groupby_and_mask(self, dtype, seed):
             },
             mask=self.build_mask(permute(rot90(self.eye_mask(shape=shape)))),
         )
+
+
+class SpecificAssetsTestCase(WithSeededRandomPipelineEngine,
```
Comment (llllllllll) [inline tests/pipeline/test_filter.py]: why does this need a seeded random loader, we are not loading any values, just sids. Can we just create an engine with an exploading object for a loader?

Comment (ssanderson) [inline tests/pipeline/test_filter.py]: We could, but setting up the engine with sids and trading days is a little bit boilerplate-ey.  I tend to use `WithSeededRandomPipelineEngine` when I just need **an** engine, but I don't care about the specific values it emits.

File: zipline/pipeline/filters/filter.py (hunk 1 above)
Comment (ssanderson) [inline zipline/pipeline/filters/filter.py]: Removed the coercions here.

**Comment by justnpT** [comment]
I would vote for adding docs about what specific assets are

**Comment by ssanderson** [comment]
@justnpT I added another sentence to the docstring explaining when you would use `SpecificAssets`: https://github.com/quantopian/zipline/pull/1530/files#diff-9bcf584f1d4b53abe9f16f81e5ee0716R503.

**Pull request merged**